### ISSUE TITLE
Support PHP 8

### ISSUE THREAD

**Issue by shyim** [api-platform/api-platform#1586]
**API Platform version(s) affected**: 2.5.6

**Description**  
Deprecated: Required parameter $operationPathResolver follows optional parameter $jsonSchemaFactory in /var/www/html/vendor/api-platform/core/src/Swagger/Serializer/DocumentationNormalizer.php on line 108

Deprecated: Required parameter $operationType follows optional parameter $type in /var/www/html/vendor/api-platform/core/src/JsonSchema/SchemaFactory.php on line 236

Deprecated: Required parameter $operationName follows optional parameter $type in /var/www/html/vendor/api-platform/core/src/JsonSchema/SchemaFactory.php on line 236

Deprecated: Required parameter $serializerContext follows optional parameter $type in /var/www/html/vendor/api-platform/core/src/JsonSchema/SchemaFactory.php on line 236

Deprecated: Required parameter $operationType follows optional parameter $type in /var/www/html/vendor/api-platform/core/src/JsonSchema/SchemaFactory.php on line 266

Deprecated: Required parameter $operationName follows optional parameter $type in /var/www/html/vendor/api-platform/core/src/JsonSchema/SchemaFactory.php on line 266

**How to reproduce**  
Use PHP 8

**Comment by christiansiewert** [comment]
@shyim Is that still up to date? I just pulled [webdevops/php-dev:8.0](https://dockerfile.readthedocs.io/en/latest/content/DockerImages/dockerfiles/php-dev.html) and for me the deprecation warnings are gone. :rocket: 

![image](https://user-images.githubusercontent.com/33094503/102694487-ac7a9200-4221-11eb-90f1-0977fb0b52a2.png)

**Comment by shyim** [comment]
Looks like will be fixed with 2.6.0

**Comment by christiansiewert** [comment]
I am using [api-platform/core v2.5.9](https://github.com/api-platform/core/releases/tag/v2.5.9) and for me it is already be fixed there:

```bash
gitpod /workspace/aaas-api $ composer show | grep api-platform/core
api-platform/core v2.5.9    Build a fully-featured hypermedia or GraphQL API in minutes!
```

I had the same warnings like you some weeks ago.

**Comment by dunglas** [comment]
Yes it has been fixed in 2.5.
